has-matching-release-tag/hasMatchingReleaseTag.ts

```diff
@@ -0,0 +1,77 @@
+import { exec, ExecOptions } from '@actions/exec'
+
+export async function hasMatchingReleaseTag(
+	refName: string,
+	releaseTagRegexp: RegExp,
+	releaseBranchWithoutPatchRegexp: RegExp,
+	releaseBranchWithPatchRegexp: RegExp,
+): Promise<string> {
+	let refNames: Array<string> = []
+	const options: ExecOptions = {
+		listeners: {
+			stdout: (data: Buffer) => {
+				refNames = data.toString().split(/(?:\r\n|\r|\n)/g)
+			},
+		},
+	}
+
+	await exec('git', ['tag'], options)
```

Comment:
Ensure proper error handling of command not found or non zero exit code from command.

Reply:
Also ensure that tags is not empty, if it is, encourage use of fetch-depth: 0

Reply:
Added in 83b5e6f and 75e01ef